### DISCUSSION TITLE
Error: WebSocket is not open: readyState 2 (CLOSING)

### DISCUSSION THREAD

**Discussion by roughnecks** [xmppjs/xmpp.js#989] [Q&A]
Hello,
I'm trying to use this: https://www.npmjs.com/package/@xmpp/client

but when using wss as a connection method, I get the error in title.
I just copy pasted the whole thing and replaced the client's variables, that's it.

I know it should not be an issue with my server because another bot is connecting via the same wss without problems.

Any help?
Thanks

```
/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/ws/lib/websocket.js:1097
    const err = new Error(
                ^

Error: WebSocket is not open: readyState 2 (CLOSING)
    at sendAfterClose (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/ws/lib/websocket.js:1097:17)
    at WebSocket.send (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/ws/lib/websocket.js:453:7)
    at Socket.write (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/@xmpp/websocket/lib/Socket.js:71:19)
    at /mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/@xmpp/connection/index.js:333:19
    at new Promise (<anonymous>)
    at Client.write (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/@xmpp/connection/index.js:325:12)
    at Client.send (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/@xmpp/connection/index.js:313:16)
    at Client.send (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/@xmpp/websocket/lib/Connection.js:22:18)
    at Client.send (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/node_modules/@xmpp/client-core/lib/Client.js:12:42)
    at Client.<anonymous> (/mnt/c/Users/roughnecks/Documents/GitHub/xmpp-ZED/zed.js:24:16)

Node.js v18.16.1
```

**Comment by sonnyp** [comment]
If you don't share the code we can't tell if you're doing something wrong.

Can you reproduce with this https://github.com/xmppjs/xmpp.js/tree/main/packages/client#example ?
Adjust configuration

**Comment by roughnecks** [comment]
> If you don't share the code we can't tell if you're doing something wrong.

Well, I used the code in the first link I posted, which is the same as yours (if I'm not mistaken).
Anyway, I just tried again with the code from your example and I can reproduce.

**Comment by sonnyp** [comment]
In general, when you need help, share as much as possible information.

What is your server / version ?
What configuration are you using?

**Comment by roughnecks** [comment]
Prosody 0.12.3

const xmpp = client({
  service: "wss://woodpeckersnest.space:5281/xmpp-websocket",
  domain: "woodpeckersnest.space",
  resource: "bot",
  username: "bot",
  password: "secret",
});

**Comment by sonnyp** [comment]
Ok try again with logs enabled using https://github.com/xmppjs/xmpp.js/tree/main/packages/debug
and share the result

**Comment by roughnecks** [comment]
> Ok try again with logs enabled using https://github.com/xmppjs/xmpp.js/tree/main/packages/debug and share the result

That was already with debug true, but if you wanted to see the whole output, it's [here](https://paste.tildeverse.org/?49a608328c1b220f#B2exHBmCv1KpfjMFQzLEnAtvG7i5Cgn1P8HZXprxb8q5)

**Comment by roughnecks** [comment]
I relogged in on my phone with the bot account and I got all of the past "hello world" messages but now I have a different error message in console: [log](https://paste.tildeverse.org/?5a9c5fc36c88670f#HcDQd9vjKWfXday3i2AJygLxpNFd2VxSPyp2myw4FeNc)

Isn't it supposed to stay connected after sending the message?

**Reply by sonnyp** [reply]
In the example code it disconnects after receiving the first message.
Comment the following:

```
xmpp.on("stanza", async (stanza) => {
  if (stanza.is("message")) {
    await xmpp.send(xml("presence", { type: "unavailable" }));
    await xmpp.stop();
  }
});
```

I'm not sure exactly what the problem is but it might be a rare bug in xmpp.js, could you file an issue with the logs?

`@xmpp/debug` already takes care of hiding the passwords but you might want to replace the `from` and `to`
